python/cudf/cudf/core/column/column.py

```diff
@@ -123,11 +124,17 @@ def mask_array_view(self):
     def __len__(self):
         return self.size
 
-    def to_pandas(self, index=None, **kwargs):
-        if str(self.dtype) in NUMERIC_TYPES and self.null_count == 0:
-            pd_series = pd.Series(cupy.asnumpy(self.values))
+    def to_pandas(self, index=None, nullable=False, **kwargs):
+        if nullable and self.dtype in cudf_dtypes_to_pandas_dtypes:
+            pandas_nullable_dtype = cudf_dtypes_to_pandas_dtypes[self.dtype]
+            arrow_array = self.to_arrow()
+            pandas_array = pandas_nullable_dtype.__from_arrow__(arrow_array)
+            pd_series = pd.Series(pandas_array, copy=False)
+        elif str(self.dtype) in NUMERIC_TYPES and self.null_count == 0:
+            pd_series = pd.Series(cupy.asnumpy(self.values), copy=False)
```

Comment:
Can we use `self.to_arrow().to_pandas()` in this situation or does that leave us with data in a bad state?

Reply:
I *think* this causes the read-only bug. https://issues.apache.org/jira/browse/ARROW-9772?page=com.atlassian.jira.plugin.system.issuetabpanels%3Aall-tabpanel

Reply:
Got it. I think the only things we need to handle datetime / timedelta types in this situation as well then.

Reply:
Added dedicated `to_pandas` implementations for datetime and timedelta columns 👍